datasets/samplemotordatalimerick.py

```diff
@@ -0,0 +1,662 @@
+        df_raw["Acceleration_z (g)"] = 50 * (df_raw["Voltage_z"] - ss_vibr_z1)
+
+        raw_data = df_raw[["Acceleration_x (g)", "Acceleration_y (g)", "Acceleration_z (g)"]]
+        raw_data = raw_data.to_numpy()
+        return ['ADXL356C', os.path.basename(file_full_path).split('/')[-1], raw_data]
+
+    def __makedir_exist_ok(self, dirpath):
+        try:
+            os.makedirs(dirpath)
+        except OSError as e:
+            if e.errno == errno.EEXIST:
+                pass
+            else:
+                raise
+
+    def __init__(self, root, d_type, transform=None,
+                 downsampling_ratio=2,
+                 signal_duration_in_sec=0.25,
+                 overlap_ratio=0.75,
+                 eval_mode=False,
+                 label_as_signal=True,
+                 random_or_speed_split=True,
+                 accel_in_second_dim=True,
+                 sensor_selected=sensor_options[0],
+                 rpm_selected=rpm_options[0]):
+
+        if d_type not in ('test', 'train'):
+            raise ValueError(
+                "d_type can only be set to 'test' or 'train'"
+                )
+
+        if rpm_selected not in SampleMotorDataLimerick.rpm_options:
+            raise ValueError(
+                f"rpm_selected can only be set from: {SampleMotorDataLimerick.rpm_options}"
+                )
+
+        if not isinstance(downsampling_ratio, int) or downsampling_ratio < 1:
+            raise ValueError(
+                "downsampling_ratio can only be set to an integer value greater than 0"
+                )
+
+        self.selected_sensor_sr = SampleMotorDataLimerick.sensor_options_sr_Hz[0]
+
+        self.root = root
+        self.d_type = d_type
+        self.transform = transform
+
+        self.downsampling_ratio = downsampling_ratio
+        self.signal_duration_in_sec = signal_duration_in_sec
+        self.overlap_ratio = overlap_ratio
+
+        self.eval_mode = eval_mode
+        self.label_as_signal = label_as_signal
+
+        self.random_or_speed_split = random_or_speed_split
+        self.accel_in_second_dim = accel_in_second_dim
+
+        self.sensor_selected = sensor_selected
+        self.rpm_selected = rpm_selected
+
+        self.num_of_features = 3
+
+        processed_folder = \
+            os.path.join(root, self.__class__.__name__, 'processed')
+
+        self.__makedir_exist_ok(processed_folder)
+
+        self.specs_identifier = f'eval_mode_{self.eval_mode}_' + \
+                                f'label_as_signal_{self.label_as_signal}_' + \
+                                f'ds_{self.downsampling_ratio}_' + \
+                                f'dur_{self.signal_duration_in_sec}_' + \
+                                f'ovlp_ratio_{self.overlap_ratio}_' + \
+                                f'random_split_{self.random_or_speed_split}_' + \
+                                f'sensor_selected_{self.sensor_selected}_' +\
```

Comment:
no need to log sensor option as identifier

Reply:
Updated